### DISCUSSION TITLE
Modify prompt regex for Arista EOS

### DISCUSSION THREAD

**Discussion by malanovo** [carlmontanari/scrapli#123] [Q&A] [answered]
Greetings all!  I ran into an issue the other day when attempting to do configuration modifications on an Arista EOS device (test device version was 4.23.0F, but it applies to others as well).  When applying changes to the TACACS config, the current regex for matching the `configuration` privilege level isn't happy about it.  Specifically, this bit of (redacted) EOS config:
```
aaa group server tacacs+ Some-Tacacs-Long-Named-Group
   server 10.0.0.1 vrf MGMT
   server 10.0.0.2 vrf MGMT
```
caused the prompt match to fail due to the length of the prompt, the usage of a "+" character, and the mixed case.  The fix looks to be a modification to the regex in `scrapli/driver/core/arista_eos/base_driver.py`, and I'd be happy to submit a patch for it.  It's possible the IOS driver will need changes as well (I'll find out shortly, I'm about to tackle that next) but I wanted to start a discussion about it in order to get some feedback first.  I was able to work around it by modifying `comms_prompt_pattern` but that wasn't ideal as it required me appending an `exit` into my configs to get the device to back itself out to a prompt the code could recognize.  My testing with modifying the regex in the aforementioned `base_driver.py` file seemed to produce a much cleaner result.

**Comment by carlmontanari** [comment]
Hey @malanovo thanks for raising this! I think(?!?) this has actually come up for the IOSXE platform but must never have been trickled back into EOS! A pr would be quite welcome for this (plus a test to cover this pretty please -- should be easy to add in [here](https://github.com/carlmontanari/scrapli/blob/master/tests/unit/driver/core/arista_eos/test_base_driver.py#L9-L39)!). 

> but that wasn't ideal as it required me appending an exit into my configs to get the device to back itself out to a prompt the code could recognize

I *think* if I'm reading ya correctly, this is not an issue when you update the priv levels in `base_driver`, correct? If so, yep that sounds right -- if you manually changed the `comms_prompt_pattern` there is a bit of other magic that would have to happen to generate the internal "map" of getting between priv levels and such. 

In any case -- a pr would be great -- let me know if there is anything I can help with, and thanks again for raising this @malanovo !

Carl

**Reply by malanovo** [reply]
Following up on this, the IOS portions of the code do not seem to have this problem, because Cisco was rational and reasonable with its prompt lengths.  For EOS, on the other hand, I wasn't able to find the upper limit to the potential prompt length.  I got up around 160 characters or so before I gave up.  My project is finishing up at this point, so hopefully within the next few days I'll have time to switch over to building out a test case and PR.

**Reply by malanovo** [reply]
Pull request (complete with test case) has been submitted against the `develop` branch.  Please let me know if there are any issues.  Thanks!

**Comment by malanovo** [comment, marked as answer]
Adding an answer to this to close it out, since the discussion about it was over in the PR.  Turns out this was fixed as a result of other changes to the code, so there wasn't a need for me to submit a PR.  Current development branch appears to not have this issue.
